crates/bevy_ecs/src/storage/blob_vec.rs

```diff
@@ -119,6 +119,19 @@ impl BlobVec {
         }
     }
 
+    /// Similar to `reserve_exact`. This method ensures that the capacity will grow at least `self.capacity()` if there is no
+    /// enough space to hold `additional` more elements.
+    pub fn reserve(&mut self, additional: usize) {
+        let available_space = self.capacity - self.len;
+        if available_space < additional && self.item_layout.size() > 0 {
+            let extra_space = self.capacity.max(additional - available_space);
+            // SAFETY: `additional - available_space > 0` so `extra_space` is non-zero
+            let increment = unsafe { NonZeroUsize::new_unchecked(extra_space) };
```

Comment:
- Let's call it `extra_capacity` because "space" can mean both "len" and "capacity"
- Also let's name both variables `extra_space` and `increment` the same, because they are the same
- Also `new_unchecked` is not really needed here, safe `new()` + `unwrap` should work equally fine
  - compiler is able to get rid of this trivial check
  - but even if it doesn't, this code is executed rarely anyway